Created gui.py file

tanya3007 opened the pull request:
Type the sheet number, browse to the file. The zip.py code will make the changes in the file and display the changed table. Also, a download button to save the file in the same directory as the gui.py script.

achoudh5 commented:
Hi @tanya3007 

Can you merge it in GUI/dashboard.py ? I kinda want both things on a single pane of GUI if that makes sense. Is it possible for you to work on it? 

PS- I haven't looked completely at `gui.py` yet unless you have already done it! Let me know :)

tanya3007 commented:
Shall I delete the current code in dashboard.py and add my code of gui.py into it ?
I have written it from scratch, so I do not know, if the code in dashboard.py is useful.

achoudh5 commented:
Ohh yeah for sure, but why I am focussing on dashboard.py is because I have something in place where I am making an api call and using that search box for that purpose. The search box should take an input like `s` in dashboard.py , just made changes to 'GUI/dashboard.py' to give you an idea.

Now, if you think you can add that feature in your new `gui.py`  by all means go for it.

tanya3007 commented:
I compiled the code in dashboard.py . Probably, there is some issue in the code. The GUI doesn't shows an input box and a button.
Can you figure it out?

Meanwhile, I'l be adding my written code for parsing excel sheet below your code.

achoudh5 commented:
Hi @tanya3007 

I am not sure why it's giving you error, mine worked fine. Can you paste the error that I can  look into ? Also, I made some changes to `readme.md` to give you an idea.

achoudh5 reviewed (changes requested):
line 18:-
`	filename = filedialog.askopenfilename(initialdir = "C:/Users/Lenovo/Desktop", title = "Select an Excel File", filetype = (("xlsx", "*.xlsx"), ("All Files", "*.*")))
`
- Can you make **initialdir** more modular? When user clicks there, they should see folders and file of their machine.
- Text is little small, can you make it a little bigger?
- Question: Is that output going to store all files processed until that time? Let's say if the user doesn't close the window throughout the day, will it give them all files they processed?

tanya3007 commented:
Hey @achoudh5 .
I **merged** the codes and they are now working fine. Added **requirements.txt**. **Initialdir** also fixed. Yes, the files will remain stored as soon as the user selects the excel file until the user terminates the GUI.

The GUI is working fine. I also modified the zip.py and valid_ip.py code a bit to work with the GUI code. 
Still, there are some other **errors** in the file **zip.py** ( not related to the GUI code ). Can you run the zip.py individually giving the parameters in __main__ and figure them out?

Can you review the GUI code?

achoudh5 reviewed (changes requested):
- I am getting this error when I click on Browse button:-

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/tkinter/__init__.py", line 1883, in __call__
    return self.func(*args)
  File "sample.py", line 49, in fileDialog
    filename = filedialog.askopenfilename(initialdir = "/", title = "Select an Excel File", filetype = (("xlsx", "*.xlsx"), ("All Files", "*.*")))
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/tkinter/filedialog.py", line 376, in askopenfilename
    return Open(**options).show()
  File "/usr/local/Cellar/python@3.8/3.8.3_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/tkinter/commondialog.py", line 44, in show
    s = w.tk.call(self.command, *w._options(self.options))
_tkinter.TclError: bad option "-filetype": must be -defaultextension, -filetypes, -initialdir, -initialfile, -message, -multiple, -parent, -title, -typevariable, or -command
```

- Also, can you align the `input`, `output` and `the other button` which was earlier there just like in 
   readme I have showcased?

tanya3007 commented:
Apologies. But I am a little bit confused regards the UI. I had a look over the readme file. 

Where exactly shall I put the Input and Output sections ?
Shall I remove that image ? and put something there?

Also, regarding that Browse File button. Is the error raised just after clicking the button, or after selecting the excel sheet? Actually, It's working fine in mine.
Probably, tkinter would be not up to date. Can you check once?

I'll change the colour of the scrollbar.

I am attaching few pictures showing how the UI is looking in my desktop.

![Screenshot (338)](https://user-images.githubusercontent.com/50319648/95019590-d7723280-0683-11eb-976e-fca91aabe4e4.png)

![Screenshot (339)](https://user-images.githubusercontent.com/50319648/95019598-de00aa00-0683-11eb-865e-da93e8392549.png)

achoudh5 commented:
> Apologies. But I am a little bit confused regards the UI. I had a look over the readme file.
> 
> Where exactly shall I put the Input and Output sections ?
> Shall I remove that image ? and put something there?
> 
> Also, regarding that Browse File button. Is the error raised just after clicking the button, or after selecting the excel sheet? Actually, It's working fine in mine.
> Probably, tkinter would be not up to date. Can you check once?
> 
> I'll change the colour of the scrollbar.
> 
> I am attaching few pictures showing how the UI is looking in my desktop.
> 
> ![Screenshot (338)](https://user-images.githubusercontent.com/50319648/95019590-d7723280-0683-11eb-976e-fca91aabe4e4.png)
> 
> ![Screenshot (339)](https://user-images.githubusercontent.com/50319648/95019598-de00aa00-0683-11eb-865e-da93e8392549.png)

- Yeah, remove that image but keep some space for an image there right above `check validity box`

- you can put the Input section on the leftmost side of the GUI i.e. before `check validity box` and Output section on the rightmost side i.e.  right after the `check validity box`

- Yeah, it's some issue with mac catolina OS for tkinter, your UI looks very impressive. Thanks for attaching the screenshots.

- Yes, it'll be great if you can change the color of the scrollbar which goes with the UI of the buttons. Also, if you can change the color of `Browse` similar to `check validity` that'll be great :)

tanya3007 commented:
Ok Sure.

I'll change the colour of Scrollbar, Browse button.
For now, I am not removing the image. You can change the image in future, since you have to add some image.

I can make the input section and Image section (consisting of Input box and Checking Validity button) in one line (inline).
But the output probably won't look good on the right side, since the columns in the excel sheet are good in number.
I am attaching a sample picture showing how the Excel sheet looks.

![Screenshot (340)](https://user-images.githubusercontent.com/50319648/95021048-48b5e380-068c-11eb-9e9e-1463c9782a53.png)

achoudh5 commented:
> Ok Sure.
> 
> I'll change the colour of Scrollbar, Browse button.
> For now, I am not removing the image. You can change the image in future, since you have to add some image.
> 
> I can make the input section and Image section (consisting of Input box and Checking Validity button) in one line (inline).
> But the output probably won't look good on the right side, since the columns in the excel sheet are good in number.
> I am attaching a sample picture showing how the Excel sheet looks.
> 
> ![Screenshot (340)](https://user-images.githubusercontent.com/50319648/95021048-48b5e380-068c-11eb-9e9e-1463c9782a53.png)

-  Actually you are right, how about we just put the `check validity` button right in line with the `browse button` pane, align them horizontally and keep some space. Below them, we can have the `Output excel file preview`, makes sense?

- Also, I found [Reference](https://stackoverflow.com/a/54849102/9420870) you can use to style buttons. 

Sample for `check validity` , this is half done. I want to change the mouse cursor to hand when someone hovers on the buttons and also, like I mentioned about color combination in previous comment. 
```
style = ttk.Style()
style.map("C.TButton",
          foreground=[('pressed', 'red'), ('active', 'blue')],
          background=[('pressed', '!disabled', 'black'),
                      ('active', 'white')]
          )
validity_Button = ttk.Button(second_frame,text="Checking validity!",command= myClick, style="C.TButton")
# canvas1.create_window(280, 180, window=validity_Button)
validity_Button.grid()
```
Let me know if this makes sense :)

tanya3007 commented:
So, first comes the Image. Then below them are the Input sections and check Validity sections. Below that, comes the Output section.
Am i right?

Rest, that cursor change, I'll do.

achoudh5 commented:
> So, first comes the Image. Then below them are the Input sections and check Validity sections. Below that, comes the Output section.
> Am i right?
> 
> Rest, that cursor change, I'll do.

**That sounds good!**

tanya3007 commented:
It is just a sample. I told you before, that there are some errors in zip.py, which can you figure out, compiling it individually.

achoudh5 commented on zip.py:
```diff
@@ -341,7 +339,7 @@ def port_number(ss):
 
 
-def main():
+def main(y, x=1):
```
```
def main():
    # a.b.c.d-> ip format
    # Sheet1 subnets

    # Use valid_ip.py in repo as module and import it in this script. Use OOPs concepts to eliminate below function

    # I am using the below two lists here which are kinda hard coded, I want to identify applications from https://applipedia.paloaltonetworks.com/ based on the input port number
    def GUI(x,y):
        global inp
        
        inp = x
        ss_inp = y
        parse_user_input(ss_inp)
        zone(ss_inp)
        port_number(ss_inp)
```

achoudh5 replied:
This change might fix the errors you were mentioning

achoudh5 replied:
I think,  you don't have to specify x as 1.

Another suggestion:-
```
x=inp
y=ss_inp
```

achoudh5 commented on GUI/dashboard.py:
```diff
@@ -1,49 +1,176 @@
 from tkinter import *
 #import <api>
 from PIL import ImageTk, Image
+
```
`import zip`

- [ ] import  zip.py as module in this script

- [ ] call zip.GUI(x,y) and put `x`= sheet number (1 or 2) & `y`= path of file

achoudh5 replied:
It would make sense if we call zip function from `gui.py` rather than reverse

tanya3007 replied:
Yes, I am doing this only. Check **Line 16** of the updated code in dashboard.py. And then called the main fucntion passing the parametes. See **Line 55** for this.

achoudh5 commented on GUI/dashboard.py:
```diff
@@ -1,49 +1,176 @@
+		sh = "Sheet"
+		sh = sh + str(e.get())
+		sheet = wb[sh]
+
```
`zip.GUI(sh,filename)` 
#this might work

tanya3007 replied:
Why have you added the funciton GUI ? I was passing the parameters to the main function before.

achoudh5 replied:
I think you took care of this 🖌️

tanya3007 replied:
Yes, I told you, I have made changes in valid_ip.py and zip.py to compensate with the dashboard.py code.

But there are some other errors in the zip.py script. They can be figured out by compiling it individually and passing the filename and sheet name in the __main__ .

tanya3007 commented:
Hey @achoudh5 ,
I did the changes as per you said, and added in the PR.

Can you review and merge it?
I am also attaching a picture of the GUI.

![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023413-91749900-069a-11eb-8152-5250da64c25d.png)

tanya3007 commented:
Hey @achoudh5 ,
I did the changes as per you said, and added in the PR.

Can you review and merge it?
I am also attaching a picture of the GUI.

![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023472-d7316180-069a-11eb-9635-585dbbe25f86.png)

achoudh5 commented:
> Hey @achoudh5 ,
> I did the changes as per you said, and added in the PR.
> 
> Can you review and merge it?
> I am also attaching a picture of the GUI.
> 
> ![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023472-d7316180-069a-11eb-9635-585dbbe25f86.png)

reviewing..

achoudh5 commented:
> > Hey @achoudh5 ,
> > I did the changes as per you said, and added in the PR.
> > Can you review and merge it?
> > I am also attaching a picture of the GUI.
> > ![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023472-d7316180-069a-11eb-9635-585dbbe25f86.png)
> 
> reviewing..

Very Well done @tanya3007 !! It looks great, one last thing I would love to have is a horizontal scroll bar as well that will simplify user experience :)

tanya3007 commented:
> > > Hey @achoudh5 ,
> > > I did the changes as per you said, and added in the PR.
> > > Can you review and merge it?
> > > I am also attaching a picture of the GUI.
> > > ![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023472-d7316180-069a-11eb-9635-585dbbe25f86.png)
> > 
> > 
> > reviewing..
> 
> * Minor fixes:
>   
>   * When I click, `Check Validity` button ideally it should print some text (take any sample text to print for your testing) it throws below error
>   * Error: `_tkinter.TclError: cannot use geometry manager grid inside . which already has slaves managed by pack`
>   * [Reference] (https://stackoverflow.com/questions/23584325/cannot-use-geometry-manager-pack-inside)
> 
> Rest Very Well done @tanya3007 !! It looks great, one last thing I would love to have is a horizontal scroll bar as well that will simplify user experience :)

There is a code in the myClick func( ). < str(s) >in line 1333. It is throwing error. 
You have written there, that it is from some api. so shall I make s="some api" for now?

achoudh5 commented:
> > > > Hey @achoudh5 ,
> > > > I did the changes as per you said, and added in the PR.
> > > > Can you review and merge it?
> > > > I am also attaching a picture of the GUI.
> > > > ![Screenshot (341)](https://user-images.githubusercontent.com/50319648/95023472-d7316180-069a-11eb-9635-585dbbe25f86.png)
> > > 
> > > 
> > > reviewing..
> > 
> > 
> > 
> > * Minor fixes:
> >   
> >   * When I click, `Check Validity` button ideally it should print some text (take any sample text to print for your testing) it throws below error
> >   * Error: `_tkinter.TclError: cannot use geometry manager grid inside . which already has slaves managed by pack`
> >   * [Reference] (https://stackoverflow.com/questions/23584325/cannot-use-geometry-manager-pack-inside)
> > 
> > Rest Very Well done @tanya3007 !! It looks great, one last thing I would love to have is a horizontal scroll bar as well that will simplify user experience :)
> 
> There is a code in the myClick func( ). < str(s) >in **line 133**. It is throwing error.
> You have written there, that it is from some api. so shall I make s="some api" for now?

`s=<text>` and ingest in 

`mylabel = Label(root, text=str(counter)+' '+'You Entered:'\
                               +input_box.get()+'\nRequestor'+' '+str(s)+'\n')`

tanya3007 commented:
I didn't got you. What you meant by "ingest in" ?
Did you meant use the text which the user enters? If yes, then it is already being used by the code < input_box.get( ) >.
Else, if you meant write a sample string for now, I am adding "some api" as the string.

Also, @achoudh5 I tried adding the horizontal scrollbar, but it is having issues. I am not able to position it properly. I am attaching a picture. Probably, we can opt it out?

![Screenshot (342)](https://user-images.githubusercontent.com/50319648/95025317-6b54f600-06a6-11eb-84c6-c6e74b48f0e2.png)

achoudh5 commented:
- Yes, use "abcd@gmail.com" rather than "some api".

- Yeah, horizontal bar can be skipped.

tanya3007 commented:
Okay Sure.
I made the changes and also marked that thing.

Can you review it?

![Screenshot (343)](https://user-images.githubusercontent.com/50319648/95025565-4bbecd00-06a8-11eb-9c8f-f973e4a13cf4.png)

achoudh5 commented:
Reviewing...

achoudh5 commented:
> Okay Sure.
> 
> I made the changes and also marked that thing.
> 
> 
> 
> Can you review it?
> 
> 
> 
> ![Screenshot (343)](https://user-images.githubusercontent.com/50319648/95025565-4bbecd00-06a8-11eb-9c8f-f973e4a13cf4.png)
> 
> 
> 
> 
> 
> 
> 
> 

Are you stretching the image in background or you have kept original image size?

tanya3007 commented:
I have changed the image size, to adjust with the left and right columns, and so that it looks better.

I have added the Change Image button.
Probably, after this, it might be daunting. lol 😆 

Can you review it?

achoudh5 commented on GUI/dashboard.py:
```diff
@@ -132,7 +132,7 @@ def myClick():
     mylabel = Label(second_frame, text=str(counter)+' '+'You Entered: '\
                                +input_box.get()+'\nRequestor: '+' '+str(s)+'\n')
-    mylabel.grid(column=3, row=6)
+    mylabel.grid(column=3, row=5+counter)
```
The latest entered name/email should be at top, it's still printing the last entered entry in the last. The number before each text i.e. counter sort of gives an idea which order the entered text got printed.

achoudh5 reviewed (changes requested):
- The text is getting printed now, congratulations! 
- The counter though needs to be tweaked

achoudh5 commented:
> I have changed the image size, to adjust with the left and right columns, and so that it looks better.
> 
> I have added the Change Image button.
> Probably, after this, it might be daunting. lol 😆
> 
> Can you review it?

Haha, I am reviewing this change. I made some comments for the last 2 commits

tanya3007 commented:
Heyy @achoudh5 ,

Did the changes. Now the first entered prints first.

Will that be okay with the whole work?

achoudh5 reviewed (commented):
This looks amazing! Unfortunately I cannot see it on MAC, I can see invisble buttons lol

Can you please upload a snapshot in images directory with some entries you checked and a sample excel file, it will be great once you complete the counter issue to make a giphy or small video showcasing what and how it happens that I can put on front page of this repo :D

Again, thank you so much for this wonderful contribution! I enjoyed it, I hope you learnt something new too :)

achoudh5 commented:
> Heyy @achoudh5 ,
> 
> Did the changes. Now the first entered prints first.
> 
> Will that be okay with the whole work?

reviewing...

tanya3007 commented:
![Screenshot (345)](https://user-images.githubusercontent.com/50319648/95026502-5466d180-06af-11eb-902f-63d680396a7b.png)
![Screenshot (346)](https://user-images.githubusercontent.com/50319648/95026504-56c92b80-06af-11eb-88fb-d07675bcc942.png)
![Screenshot (347)](https://user-images.githubusercontent.com/50319648/95026506-57fa5880-06af-11eb-8504-665d7cb99321.png)

achoudh5 commented on GUI/dashboard.py:
```diff
@@ -139,14 +139,20 @@ def chng_img():
+    for i in range(5, 5+counter):
+
+	    mylabel = Label(second_frame, text=lis[i-5])
+	    mylabel.grid(column=3, row=i+1)
```
It's still printing in increasing order :/

tanya3007 replied:
Check now.. I resolved it..

Changed the < i-5 > to < 4-i >

achoudh5 commented:
> Heyy,
> Probably, You can use images from here.
> I think I have completed the task given to me. 😄
> Yess, I got to learn a lot working on this repo.
> 
> ![Screenshot (345)](https://user-images.githubusercontent.com/50319648/95026502-5466d180-06af-11eb-902f-63d680396a7b.png)
> ![Screenshot (346)](https://user-images.githubusercontent.com/50319648/95026504-56c92b80-06af-11eb-88fb-d07675bcc942.png)
> ![Screenshot (347)](https://user-images.githubusercontent.com/50319648/95026506-57fa5880-06af-11eb-8504-665d7cb99321.png)

Sounds good! I'll merge it

tanya3007 commented:
Hey @achoudh5 ,
Thanks for merging this.
The task was tedious, but definitely learnt a lot!.

Also, I made another PR, and now it's printing in decreasing order. You can merge it.

achoudh5 commented:
I made it tedious on fly :P

This is real world scenarios, good luck for your future!

tanya3007 commented:
Haha !
Thanks a lot... 🔥 

PS:- Don't forget merging that another PR lol. Had a bad timing due to which have to make another. :P
